moveit_ros/benchmarks/src/BenchmarkExecutor.cpp

```diff
@@ -897,6 +922,111 @@ void BenchmarkExecutor::collectMetrics(PlannerRunData& metrics,
   }
 }
 
+void BenchmarkExecutor::computeResultPathSimilarity(
+    PlannerBenchmarkData& planner_data, const std::vector<planning_interface::MotionPlanDetailedResponse>& mp_res,
+    const std::vector<bool>& solved)
+{
+  ROS_INFO("Computing result path similarity");
+  const size_t result_count = planner_data.size();
+  size_t unsolved = std::count_if(solved.begin(), solved.end(), [](bool s) { return !s; });
+  std::vector<double> average_distances(mp_res.size());
+  for (size_t first_traj_i = 0; first_traj_i < result_count; ++first_traj_i)
+  {
+    // If trajectory was not solved there is no valid average distance so it's set to max double only
+    if (!solved[first_traj_i])
+    {
+      average_distances[first_traj_i] = std::numeric_limits<double>::max();
+      continue;
+    }
+    // Iterate all result trajectories that haven't been compared yet
+    for (size_t second_traj_i = first_traj_i + 1; second_traj_i < result_count; ++second_traj_i)
+    {
+      // Ignore if other result has not been solved
+      if (!solved[second_traj_i])
+        continue;
+
+      // Abort if there are no result trajectories
+      if (mp_res[first_traj_i].trajectory_.empty() || mp_res[second_traj_i].trajectory_.empty())
```

Comment:
Did you ever run into empty trajectory lists (or empty trajectories) when the runs were marked as solved? If so that points to a bug somewhere else in the planners.

Reply:
Actually I didn't. I can remove this line if you want